plugins/logs/plugin_test.go

```diff
@@ -482,6 +483,284 @@ func TestPluginRequeBufferPreserved(t *testing.T) {
+
+	bytesWritten := fixture.plugin.enc.bytesWritten
+	if bytesWritten == 0 {
+		t.Fatal("Expected event to be written into the encoder")
+	}
+
+	_ = fixture.plugin.Log(ctx, event2) // event 2 should not be written into the encoder as rate limit exceeded
+
+	if fixture.plugin.enc.bytesWritten != bytesWritten {
+		t.Fatalf("Expected %v bytes written into the encoder but got %v", bytesWritten, fixture.plugin.enc.bytesWritten)
+	}
+
+	time.Sleep(1 * time.Second)
+	_ = fixture.plugin.Log(ctx, event2) // event 2 should now be written into the encoder
+
+	if fixture.plugin.buffer.Len() != 1 {
+		t.Fatalf("Expected buffer length of 1 but got %v", fixture.plugin.buffer.Len())
+	}
+
+	exp := EventV1{
+		Labels: map[string]string{
+			"id":      "test-instance-id",
+			"app":     "example-app",
+			"version": version.Version,
+		},
+		DecisionID:  "abc",
+		Path:        "foo/bar",
+		Input:       &input,
+		Result:      &result,
+		RequestedBy: "test-1",
+		Timestamp:   ts,
+	}
+	compareLogEvent(t, fixture.plugin.buffer.Pop(), exp)
+
+	chunk, err := fixture.plugin.enc.Flush()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	exp = EventV1{
+		Labels: map[string]string{
+			"id":      "test-instance-id",
+			"app":     "example-app",
+			"version": version.Version,
+		},
+		DecisionID:  "def",
+		Path:        "foo/baz",
+		Input:       &input,
+		Result:      &result,
+		RequestedBy: "test-2",
+		Timestamp:   ts,
+	}
+
+	compareLogEvent(t, chunk, exp)
+}
+
+func TestPluginRateLimitFloat(t *testing.T) {
+	ctx := context.Background()
+
+	ts, err := time.Parse(time.RFC3339Nano, "2018-01-01T12:00:00.123456Z")
+	if err != nil {
+		panic(err)
+	}
+
+	numDecisions := 0.1 // 0.1 decision per second ie. 1 decision per 10 seconds
+
+	fixture := newTestFixture(t, func(c *Config) {
+		limit := float64(numDecisions)
+		c.Reporting.MaxDecisionsPerSecond = &limit
+	}, func(c *Config) {
+		limit := int64(300)
+		c.Reporting.UploadSizeLimitBytes = &limit
+	})
+	defer fixture.server.stop()
+
+	var input interface{} = map[string]interface{}{"method": "GET"}
+	var result interface{} = false
+
+	event1 := &server.Info{
+		DecisionID: "abc",
+		Path:       "foo/bar",
+		Input:      &input,
+		Results:    &result,
+		RemoteAddr: "test-1",
+		Timestamp:  ts,
+	}
+
+	event2 := &server.Info{
+		DecisionID: "def",
+		Path:       "foo/baz",
+		Input:      &input,
+		Results:    &result,
+		RemoteAddr: "test-2",
+		Timestamp:  ts,
+	}
+
+	_ = fixture.plugin.Log(ctx, event1) // event 1 should be written into the encoder
+
+	bytesWritten := fixture.plugin.enc.bytesWritten
+	if bytesWritten == 0 {
+		t.Fatal("Expected event to be written into the encoder")
+	}
+
+	_ = fixture.plugin.Log(ctx, event2) // event 2 should not be written into the encoder as rate limit exceeded
+
+	if fixture.plugin.enc.bytesWritten != bytesWritten {
+		t.Fatalf("Expected %v bytes written into the encoder but got %v", bytesWritten, fixture.plugin.enc.bytesWritten)
+	}
+
+	time.Sleep(5 * time.Second)
```

Comment:
Is there no way to avoid this? Adding a 5 second sleep into the test run is suboptimal...

Reply:
If there isn't we should add the `slow` build tag that is currently used to differentiate slow tests, I suppose.

Reply:
I don't think we can avoid that if we plan to keep the test case. I've added a build tag as Stephan suggested.